doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -195,19 +285,548 @@ Mail chapel-developers@ for review and merge
+    ssh git@github.com
+
+    # Clone your fork of the chapel repo!
+    git clone git@github.com:<github_username>/chapel.git
+
+    # Or, if using HTTPS instead of SSH.
+    git clone https://github.com/<github_username>/chapel.git
+
+    # Set up remotes for github
+    cd chapel
+    git remote add upstream https://github.com/chapel-lang/chapel.git
+    # Make sure it works, get up-to-date without modifying your files
+    git fetch upstream
+    # Change remote for upstream push to "no_push" 
+    git remote set-url --push upstream no_push 
+    # Optionally add remotes for commonly viewed branches
+    git remote add <branch_owner_username> https://github.com/<branch_owner_username>/chapel.git
+
+.. _New branch command:
+
+New branch command
+++++++++++++++++++
+
+.. code-block:: bash
+
+    git checkout -b <branch_name>
+
+.. _Development commands:
+
+Development commands
+++++++++++++++++++++
+
+Stage a file/dir for commit:
+
+.. code-block:: bash
+
+    git add path/to/file
+
+    # (sort of) similar to:
+    svn add path/to/file
+
+Delete a file/dir and stage the change for commit:
+
+.. code-block:: bash
+
+    git rm [-r] path/to/dir/or/file
+
+    # similar to:
+    svn delete path/to/dir/or/file
+
+Move a file/dir:
+
+.. code-block:: bash
+
+    git mv orig/path/a.txt new/path/to/b.txt
+
+    # similar to:
+    svn move orig/path/a.txt new/path/to/b.txt
+
+Copy a file/dir and stage target for commit:
+
+.. code-block:: bash
+
+    cp <src> <target>
+    git add <target>
+
+    # similar to:
+    svn copy <src> <target>
+
+Get the status of files/dirs (staged and unstaged):
+
+.. code-block:: bash
+
+    git status
+
+    # similar to:
+    svn status
+
+Get the diff of unstaged changes:
+
+.. code-block:: bash
+
+    git diff
+
+    # similar to:
+    svn diff
+
+Get the diff of staged changes (those that were staged with ``git add``):
+
+.. code-block:: bash
+
+    git diff --cached
+
+Backing out unstaged changes:
+
+.. code-block:: bash
+
+    git checkout path/to/file/a.txt
+
+    # similar to:
+    svn revert path/to/file/a.txt
+
+Committing staged changes:
+
+.. code-block:: bash
+
+    git commit [-m <message>]
+
+    # similar to:
+    svn commit [-m <message>]
+
+Bring in changes from the main Chapel project:
+
+.. code-block:: bash
+
+    git fetch upstream
+    git merge upstream/master
+
+    # or:
+    git pull upstream <branch_name>
+
+    # with feature branch checked out:
+    git merge [--no-ff] upstream/master
+
+If there are conflicts, you will be asked to resolve them. Once the affected
+files have been fixed, stage them with ``git add``, and then call ``git
+commit`` to finish the merge process.
+
+If you want to understand the changes that occurred upstream, see
+`Read commit messages`_ below.
+
+Fixing a commit message:
+
+.. code-block:: bash
+
+    git commit --amend
+
+Note: This should only ever be done to commits that **have not been pushed** to
+a remote repository.
```

Comment:
I actually disagree about this. It's OK to do with commits that have only been pushed to your feature branch. I do it regularly that way. You just have to `git push -f` when you send them up to GitHub.